### PR TITLE
change the library from net to netip for integration framework components

### PR DESCRIPTION
**Pull request by zhlsunshine**
**Please provide a description of this PR:**
change the library from net to netip for integration framework components.
this PR is related to issue #39348 

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [ ] Ambient
- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

**Please check any characteristics that apply to this pull request.**

- [ ] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

File: pkg/test/framework/components/istio/ingress.go [hunk 1]
```diff
@@ -156,14 +157,17 @@ func (c *ingressImpl) HTTPSAddress() (string, int) {
...
-func (c *ingressImpl) DiscoveryAddress() net.TCPAddr {
+func (c *ingressImpl) DiscoveryAddress() netip.AddrPort {
 	host, port := c.AddressForPort(discoveryPort)
-	ip := net.ParseIP(host)
-	if ip.String() == "<nil>" {
+	ip, err := netip.ParseAddr(host)
+	if err != nil {
+		return netip.AddrPort{}
+	}
+	if !ip.IsValid() {
```
Comment (howardjohn) [inline pkg/test/framework/components/istio/ingress.go:166]: can ParseAddr really return an invalid IP and err=nil?

**Review by howardjohn — APPROVED** [review]

File: pkg/test/framework/components/istio/ingress.go (hunk 1 above)
Comment (zhlsunshine) [reply pkg/test/framework/components/istio/ingress.go:166]: no, got it, I will remove the `IsValid` check.

**Pull request merged**